test/mitmproxy/addons/test_xss.py

```diff
@@ -0,0 +1,280 @@
+import unittest
+from unittest import mock
+from mitmproxy.addons import xss
+
+
+class MockResponseOrRequest:
+    def __init__(self, cookies):
+        self.cookies = MockLoCT(cookies)
+        self.content = "<html></html>"
+        self.url = "https://example.com/index.html?q=1"
+
+
+class MockLoCT:
+    def __init__(self, cookies):
+        self.fields = cookies
+
+
+class MockCookieValue:
+    def __init__(self, value):
+        self.value = value
+
+
+class MockFlow:
```

Comment:
Look at the tests in test/mitmproxy/ (test_example.py might be interesting in particular) - you should be able to use regular mitmproxy flows here.